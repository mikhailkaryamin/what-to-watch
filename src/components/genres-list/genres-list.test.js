import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';

import {
  initialState,
} from '../../mocks/initialState.js';

import GenresList from './genres-list.jsx';

const mockStore = configureStore([]);

describe(`Genres list`, () => {
  const store = mockStore(initialState);
  const onGenreClick = () => {};

  const wrapper = renderer
    .create(
        <Provider store={store}>
          <GenresList onGenreClick={onGenreClick}/>
        </Provider>
    );

  test(`should render genres list`, () => {
    expect(wrapper).toMatchSnapshot();
  });
});
