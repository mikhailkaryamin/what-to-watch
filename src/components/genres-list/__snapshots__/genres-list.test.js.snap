// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Genres list should render genres list 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Comedies
    </a>
  </li>
  <li
    className="catalog__genres-item"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Crime
    </a>
  </li>
  <li
    className="catalog__genres-item"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Documentary
    </a>
  </li>
  <li
    className="catalog__genres-item"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Dramas
    </a>
  </li>
  <li
    className="catalog__genres-item"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Horror
    </a>
  </li>
  <li
    className="catalog__genres-item"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Kids & Family
    </a>
  </li>
  <li
    className="catalog__genres-item"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Romance
    </a>
  </li>
  <li
    className="catalog__genres-item"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Sci-Fi
    </a>
  </li>
  <li
    className="catalog__genres-item"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Thrillers
    </a>
  </li>
</ul>
`;
